web-server/src/hooks/useCountUp.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from 'react';
+
+export const useCountUp = (
+  targetValue: number,
+  duration: number = 1000,
+  decimalPlaces: number = 0
+): number => {
+  const [count, setCount] = useState<number>(0);
+
+  useEffect(() => {
+    let start = 0;
+    const increment = targetValue / (duration / 16);
```

Comment:
Why 16? Prefer not using magic numbers.
Fixing them can be as simple as declaring them as a constant somewhere.

In this case, right outside the hook might be fine.